controllers/extensionRequests.js

```diff
@@ -202,12 +202,23 @@ const getSelfExtensionRequests = async (req, res) => {
  * @param res {Object} - Express response object
  */
 const updateExtensionRequest = async (req, res) => {
+  const { dev } = req.query;
+  const isDev = dev === "true";
   try {
     const extensionRequest = await extensionRequestsQuery.fetchExtensionRequest(req.params.id);
     if (!extensionRequest.extensionRequestData) {
       return res.boom.notFound("Extension Request not found");
     }
 
+    if (
+      isDev &&
+      !req.userData?.roles.super_user &&
+      (extensionRequest.extensionRequestData.status === EXTENSION_REQUEST_STATUS.APPROVED ||
+        extensionRequest.extensionRequestData.status === EXTENSION_REQUEST_STATUS.DENIED)
```

Comment:
- instead of checking for status not being approved (or) denied can we check for pending directly? so that if any new status is added in the future it does not slip through this check.

Reply:
Changed the condition to check for pending directly.